File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer,Integer>map = new HashMap<>();
        for(int num:nums) map.put(num,map.getOrDefault(num, 0)+1);
        PriorityQueue<Integer>pq = new PriorityQueue<>((a,b)->map.get(b)-map.get(a));
        for(int key:map.keySet()) pq.add(key);
        int[] ans = new int[k];
        int i=0;
        while(!pq.isEmpty() && k-- >0){
            ans[i++]=pq.poll();
        }
        return ans;
    }
}

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
       unordered_map<int,int>map;
       vector<int>v(k);
       for(int num:nums){
         map[num]++;
       }
       priority_queue<pair<int,int>,vector<pair<int,int>>,greater<pair<int,int>>>pq;
       for(auto it:map){
        pq.push({it.second,it.first});
        if(pq.size()>k){
            pq.pop();
        }
       }
       while(!pq.empty()){
        v[--k]=pq.top().second;
        pq.pop();
       }
       return v;

    }
};